shell/common/shell_test.cc

```diff
@@ -101,6 +101,41 @@ void ShellTest::RunEngine(Shell* shell, RunConfiguration configuration) {
   latch.Wait();
 }
 
+void ShellTest::RestartEngine(Shell* shell, RunConfiguration configuration) {
+  fml::AutoResetWaitableEvent latch;
+  fml::TaskRunner::RunNowOrPostTask(
+      shell->GetTaskRunners().GetUITaskRunner(),
+      [shell, &latch, &configuration]() {
+        bool restarted = shell->engine_->Restart(std::move(configuration));
```

Comment:
fyi: this would be a good place to use std::future<bool>

Reply:
Done.

Reply:
I was thinking that the value of the future would be `restarted` then you can have the assert inside the test thread: ASSERT_TRUE(restarted.get_future().wait()).